packages/features/calendar-cache-sql/CalendarSubscriptionService.ts

```diff
@@ -0,0 +1,92 @@
+import { getCalendar } from "@calcom/app-store/_utils/getCalendar";
+import logger from "@calcom/lib/logger";
+import { safeStringify } from "@calcom/lib/safeStringify";
+import type { CredentialForCalendarService } from "@calcom/types/Credential";
+
+import type {
+  GoogleChannelProps,
+  ICalendarSubscriptionService,
+} from "./CalendarSubscriptionService.interface";
+
+const log = logger.getSubLogger({ prefix: ["CalendarSubscriptionService"] });
+
+const ONE_MONTH_IN_MS = 30 * 24 * 60 * 60 * 1000;
+const GOOGLE_WEBHOOK_URL_BASE = process.env.NEXT_PUBLIC_WEBAPP_URL;
+const GOOGLE_WEBHOOK_URL = `${GOOGLE_WEBHOOK_URL_BASE}/api/webhook/google-calendar-sql`;
+
+export class CalendarSubscriptionService implements ICalendarSubscriptionService {
+  async watchCalendar(
+    calendarId: string,
+    credential: CredentialForCalendarService
+  ): Promise<GoogleChannelProps | undefined> {
+    log.debug("watchCalendar", safeStringify({ calendarId }));
+
+    if (!process.env.GOOGLE_WEBHOOK_TOKEN) {
+      log.warn("GOOGLE_WEBHOOK_TOKEN is not set, skipping watching calendar");
+      return;
+    }
+
+    try {
+      const calendarService = await getCalendar(credential);
+
+      if (!calendarService) {
+        throw new Error("Could not get calendar service");
+      }
+
+      if (!calendarService.watchCalendar) {
+        throw new Error("Calendar service does not support watching calendars");
+      }
+
+      const watchResult = await calendarService.watchCalendar({
+        calendarId,
+        eventTypeIds: [],
+      });
```

Comment:
DevinAI, we're calling the old calendarSerivce inside of this new CalendarSubscriptionService. Because of this, we're subscribing to the old endpoint instead of the new webhook enpoint. I want CalendarSubscriptionService to be a standalone service. It should not depend on the old CalendarService at all. It should handle it's own initialization logic and fetcher methods. Importing googleaplis directly in there. There should not be any dependecy on the current calendarservice. This is an opportunity to come up with better and cleaner patterns and avoiding the getCalendar helper at all.